Runtime/BacktraceClient.cs

```diff
@@ -629,12 +627,22 @@ private BacktraceData SetupBacktraceData(BacktraceReport report)
 
             report.AssignSourceCodeToReport(sourceCode);
 
-            var reportAttributes = _nativeClient == null
-                ? _clientAttributes
-                : _nativeClient.GetAttributes().Merge(_clientAttributes);
-
             // pass copy of dictionary to prevent overriding client attributes
-            return report.ToBacktraceData(new Dictionary<string, string>(reportAttributes), GameObjectDepth);
+            var result = report.ToBacktraceData(null, GameObjectDepth);
+
+            // add native attributes to client report
+            if (_nativeClient != null)
+            {
+                _nativeClient.GetAttributes(result.Attributes.Attributes);
+            }
+
+            // apply client attributes
+            foreach (var attribute in _clientAttributes)
+            {
+                result.Attributes.Attributes[attribute.Key] = attribute.Value;
```

Comment:
Merge call was an extension method in c# that were merging two dictionaries by creating new one and assigning all items from one resource to another. This causes unnecessary allocations. Right now instead of recreating dictionary and working on a copy, we're appending data to existing dictionary that exists only for current method. Because of that we avoid making a copy of a dictionary and copy of each entry